src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useMemo, useState } from 'react';
+
+import FilterSelect from '@catalog/CreateFromInstanceTypes/components/AddBootableVolumeModal/components/FilterSelect/FilterSelect';
+import {
+  DEFAULT_INSTANCETYPE_LABEL,
+  DEFAULT_PREFERENCE_LABEL,
+} from '@catalog/CreateFromInstanceTypes/utils/constants';
+import {
+  VirtualMachineClusterInstancetypeModelGroupVersionKind,
+  VirtualMachineClusterPreferenceModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import DataSourceModel from '@kubevirt-ui/kubevirt-api/console/models/DataSourceModel';
+import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { V1alpha2VirtualMachineClusterPreference } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { convertResourceArrayToMap } from '@kubevirt-utils/resources/shared';
+import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import { k8sPatch, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { Form, FormGroup, Select, TextArea } from '@patternfly/react-core';
+
+import { getPreferenceReadableOS } from '../../utils/utils';
+
+type BootableVolumeMetadata = {
+  labels: { [key: string]: string };
+  annotations: { [key: string]: string };
+};
+
+type EditBootSourceModalProps = {
+  dataSource: V1beta1DataSource;
+  isOpen: boolean;
+  onClose: () => void;
+  preferences: V1alpha2VirtualMachineClusterPreference[];
+  instanceTypesNames: string[];
+};
+
+const EditBootableVolumesModal: FC<EditBootSourceModalProps> = ({
+  dataSource,
+  isOpen,
+  onClose,
+  preferences,
+  instanceTypesNames,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [, setIsDropdownOpen] = React.useState(false);
+
+  const preferencesMap = useMemo(() => convertResourceArrayToMap(preferences), [preferences]);
+  const preferencesNames = Object.keys(preferencesMap);
```

Comment:
preferencesMap is used only here - u can enter it into the useMemo above and save unneeded calculations

Reply:
Now I can see! Thanks!